packages/salesforcedx-apex-debugger/src/core/breakpointService.ts

```diff
@@ -104,18 +104,25 @@ export class BreakpointService {
         .build(),
       { cwd: projectPath }
     ).execute();
-    const result = await this.getIdFromCommandResult(execution);
-    if (this.isApexDebuggerBreakpointId(result.getId())) {
-      return Promise.resolve(result);
-    } else {
-      return Promise.reject(result.getStdOut());
+
+    const cmdOutput = new CommandOutput();
+    const result = await cmdOutput.getCmdResult(execution);
+    try {
+      const breakpointId = JSON.parse(result).result.id as string;
+      if (this.isApexDebuggerBreakpointId(breakpointId)) {
+        return Promise.resolve(breakpointId);
+      } else {
+        return Promise.reject(result);
+      }
+    } catch (e) {
+      return Promise.reject(e);
```

Comment:
It looks you want this to catch problems with the JSON.parse and throw the "result" which came from stdout. Is that what we have in "e"?

Reply:
e is a SyntaxError. The change to streamingService.test.ts shows what the format is. Do you prefer having the raw json so you can see what was attempted to be parsed?

Reply:
Yea, "Unexpected token n in JSON at position 2" is not very useful.